client/src/Components/CommonComponents/OfferCard/index.jsx

```diff
@@ -1,6 +1,128 @@
 import React from 'react';
-import UnderConstruction from '../../UnderConstruction';
+import { Card, Button, Spinner } from 'react-bootstrap';
+import { withRouter } from 'react-router-dom';
+import PropTypes from 'prop-types';
 
-export default function OfferCard() {
-  return <UnderConstruction />;
+import './style.css';
+
+class OfferCard extends React.Component {
+  state = {
+    offer: null,
+    statusLabel: '',
+    statusDiv: '',
+    hovered: '',
+    saved: false,
+  };
+
+  componentDidMount() {
+    const { offer, status } = this.props;
+    const borderColor = {
+      completed: 'green',
+      finished: 'red',
+      pending: 'orange',
+      active: 'blue',
+    };
+    this.setState(() => {
+      return {
+        offer,
+        saved: offer.saved,
+        statusLabel: `offer-card__status--${status}`,
+        statusDiv: `offer-card__border--${borderColor[status]}`,
+      };
+    });
+  }
+
+  savedClassStatus = () => {
+    const { saved } = this.state;
+    if (saved) return 'offer-card__favourite--saved';
```

Comment:
suggestion: you can use ternary condition for shortening code
`
return(saved?'offer-card__favourite--saved':'offer-card__favourite--saved');
`